charms/tensorboards-web-app/CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+# Contributing
+
+## Overview
+
+This document explains the processes and practices recommended for contributing enhancements to
+this operator.
+
+- Generally, before developing enhancements to this charm, you should consider [opening an issue
+  ](https://github.com/canonical/kubeflow-tensorboards-operator/issues) explaining your use case.
+- If you would like to chat with us about your use-cases or proposed implementation, you can reach
+  us at [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev) or [Discourse](https://discourse.charmhub.io/).
+- Familiarising yourself with the [Charmed Operator Framework](https://juju.is/docs/sdk) library will help you a lot when working on new features or bug fixes.
+- All enhancements require review before being merged. Code review typically examines
+  - code quality
+  - test coverage
+  - user experience for Juju administrators of this charm.
+- Please help us out in ensuring easy to review branches by rebasing your pull request branch onto
+  the `main` branch. This also avoids merge commits and creates a linear Git commit history.
+
+## Developing
+
+You can use the environments created by `tox` for development:
+
+```shell
+tox --notest -e unit
+source .tox/unit/bin/activate
+```
+
+### Testing
+
+```shell
+tox -e lint          # code style
+tox -e unit          # unit tests
+tox -e integration   # integration tests
+tox                  # runs 'lint' and 'unit' environments
+```
+
+## Build charm
+
+Build the Tensorboards Web App charm using:
+
+```shell
+charmcraft pack
+```
+
+## Upgrade manifests
+The charm uses Jinja2 templates in order to store manifests that need to be applied during its deployment. Those manifests must point to their target version and thus, there's a case they need to be upgraded. The process for upgrading them is:
```

Comment:
True that it could be omitted. My intention was to add some more context. What I meant to say is that when we 're deploying a new release of KF, the manifests must point to KF's new release manifests, which though could be considered common sense. I'm with removing the sentence if there's no objection.

Reply:
I see what you mean and think it comes through clear without it